drivers/telescope/lx200classic.cpp

```diff
@@ -325,3 +346,226 @@ bool LX200Classic::saveConfigItems(FILE *fp)
+
+    ln_lnlat_posn observer;
+    observer.lat = LocationN[LOCATION_LATITUDE].value;
+    observer.lng = LocationN[LOCATION_LONGITUDE].value;
+    if (observer.lng > 180)
+        observer.lng -= 360;
+
+    ln_hrz_posn horizontalPos;
+    // Libnova south = 0, west = 90, north = 180, east = 270
+
+    horizontalPos.az = parkAz + 180;
+    if (horizontalPos.az > 360)
+        horizontalPos.az -= 360;
+    horizontalPos.alt = parkAlt;
+
+    ln_equ_posn equatorialPos;
+
+    ln_get_equ_from_hrz(&horizontalPos, &observer, ln_get_julian_from_sys(), &equatorialPos);
+
+    if (isSimulation())
+    {
+        currentRA = equatorialPos.ra / 15.0;
+        currentDEC= equatorialPos.dec;
+    }
+    else
+    {
+        if ((setObjectRA(PortFD, equatorialPos.ra / 15.0) < 0) || (setObjectDEC(PortFD, equatorialPos.dec) < 0))
+        {
+            LOG_ERROR("Error setting Unpark RA/Dec.");
+            return false;
+        }
+
+        char syncString[256];
+        if (::Sync(PortFD, syncString) < 0)
+        {
+            LOG_WARN("Unpark Sync failed.");
+            return false;
+        }
+    }
+
+    SetParked(false);
+    return true;
+}
+
+bool LX200Classic::SetCurrentPark()
+{
+    ln_hrz_posn horizontalPos;
+    // Libnova south = 0, west = 90, north = 180, east = 270
+
+    ln_lnlat_posn observer;
+    observer.lat = LocationN[LOCATION_LATITUDE].value;
+    observer.lng = LocationN[LOCATION_LONGITUDE].value;
+    if (observer.lng > 180)
+        observer.lng -= 360;
+
+    ln_equ_posn equatorialPos;
+    equatorialPos.ra  = currentRA * 15;
+    equatorialPos.dec = currentDEC;
+    ln_get_hrz_from_equ(&equatorialPos, &observer, ln_get_julian_from_sys(), &horizontalPos);
+
+    double parkAZ = horizontalPos.az - 180;
+    if (parkAZ < 0)
+        parkAZ += 360;
+    double parkAlt = horizontalPos.alt;
+
+    char AzStr[16], AltStr[16];
+    fs_sexa(AzStr, parkAZ, 2, 3600);
+    fs_sexa(AltStr, parkAlt, 2, 3600);
+
+    LOGF_DEBUG("Setting current parking position to coordinates Az (%s) Alt (%s)...", AzStr, AltStr);
+
+    SetAxis1Park(parkAZ);
+    SetAxis2Park(parkAlt);
+
+    return true;
+}
+
+bool LX200Classic::SetDefaultPark()
+{
+    // Az = 0 for North hemisphere
+    SetAxis1Park(LocationN[LOCATION_LATITUDE].value > 0 ? 0 : 180);
+
+    // Alt = Latitude
+    SetAxis2Park(LocationN[LOCATION_LATITUDE].value);
+
+    return true;
+}
+
+bool LX200Classic::ReadScopeStatus()
+{
+    if (!isConnected())
+        return false;
+
+    if (isSimulation())
+    {
+        mountSim();
+        return true;
+    }
+
+    if ((TrackState == SCOPE_PARKING) &&  (isSlewComplete()))
+    {
+        setAlignmentMode(PortFD, LX200_ALIGN_LAND);
+        
+        //IUResetSwitch(&AlignmentSP);
+        //AlignmentSP.s = IPS_OK;
+        //AlignmentS[LX200_ALIGN_LAND].s = ISS_ON;
+        //IDSetSwitch(&AlignmentSP, nullptr);
+    }
+
+    LX200Generic::ReadScopeStatus();
```

Comment:
I think that you have to call the base class method first -- there are code branches that could return false from there.

Reply:
What a nice coincidence!

You are right on this. I think the two alternate exit paths are already included in my override, but it is not safe. Calling first the base class spoils the state that I need to query. 

I have seen a full code clone of the base class in another example. I tried to avoid that. I wil take a further look on this code.

Reply:
hmm, I didn't think about it spoiling my state... let me look that over some more in my code